File: src/main/java/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/TargetFileType.java
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes;

import com.intellij.openapi.fileTypes.LanguageFileType;
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileIcon;
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileLanguage;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.Icon;

public class TargetFileType extends LanguageFileType {
  public static final TargetFileType INSTANCE = new TargetFileType();

  private TargetFileType() {
    super(UnitFileLanguage.INSTANCE);
  }

  @NotNull
  @Override
  public String getName() {
    return "Target unit configuration for systemd";
  }

  @NotNull
  @Override
  public String getDescription() {
    return "A unit configuration file whose name ends in \".target\" encodes information about a target unit of systemd, which is used "
           + "for grouping units and as well-known synchronization points during start-up.";
  }

  @NotNull
  @Override
  public String getDefaultExtension() {
    return "target";
  }

  @Nullable
  @Override
  public Icon getIcon() {
    return UnitFileIcon.FILE;
  }
}

File: src/main/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/TargetFileType.kt
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes

import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileIcon
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileLanguage
import org.jetbrains.annotations.Nls
import javax.swing.Icon

class TargetFileType private constructor() : AbstractUnitFileType(UnitFileLanguage.INSTANCE) {
  override fun getName(): String {
    return "Target unit configuration for systemd"
  }

  override fun getDescription(): String {
    return displayName
  }

  override fun getDefaultExtension(): String {
    return "target"
  }

  override fun getIcon(): Icon? {
    return UnitFileIcon.FILE
  }

  override fun getDisplayName(): @Nls String {
    return "Target unit configuration (systemd)"
  }

  companion object {
    val INSTANCE = TargetFileType()
  }
}
